modules/zip/src/hash-file-utility.ts

```diff
@@ -43,6 +43,11 @@ export async function makeHashTableFromZipHeaders(
   const hashTable: Record<string, bigint> = {};
 
   for await (const cdHeader of zipCDIterator) {
+    // Some archivers add records with size=0 for directories,
+    // so we remove it in order to shorten the hash info and fastern the search
+    if (cdHeader.compressedSize === 0n) {
```

Comment:
Nit: Is there a link to some additional information that compressedSize = 0 is a special case?  This looks somewhat ad-hoc and may not be completely clear / convincing to a maintainer.

Reply:
I gave it a thought and decided to remove it completely because compressedSize === 0 can be true not only for directory records and I can't figure out fast what's the correct way to detect them... So I'd better let it be for now